contrib/golang/filters/network/source/go/pkg/network/filter.go

```diff
@@ -55,6 +56,14 @@ func (n *connectionCallback) StreamInfo() api.StreamInfo {
 	return n
 }
 
+func (n *connectionCallback) EnableHalfClose(enabled bool) {
+	var enableHalfCloseInt int
+	if enabled {
+		enableHalfCloseInt = 1
+	}
+	n.connEnableHalfCloseFunc(n.wrapper, enableHalfCloseInt)
```

Comment:
better panic with an friendly message when `connEnableHalfCloseFunc` is nil (it's a downstream connection).
i.e. `EnableHalfClose is not supported for downstream connection yet`

Reply:
done